catena4470_test01/catena4470_test01.ino

```diff
@@ -0,0 +1,1039 @@
+ *  drive the template based on the type of Serial1,
+ *  eliminating a complex and never-exhaustive series of
+ *  #ifs.
+ */
+cCatenaModbusRtu host(0, A4); // this is host and RS-232 or USB-FTDI
+ModbusSerial<decltype(Serial1)> mySerial(&Serial1);
+
+#define kPowerOn	A3
+
+//
+// the LoRaWAN backhaul.  Note that we use the
+// Catena version so it can provide hardware-specific
+// information to the base class.
+//
+Catena::LoRaWAN gLoRaWAN;
+
+// the LED instance object
+StatusLed gLed (Catena::PIN_STATUS_LED);
+
+// the RTC instance, used for sleeping
+#ifdef ARDUINO_ARCH_SAMD
+CatenaRTC gRtc;
+#elif defined(ARDUINO_ARCH_STM32)
+CatenaStm32L0Rtc gRtc;
+#endif
+
+// The BME280 instance, for the temperature/humidity sensor
+Adafruit_BME280 gBME280; // The default initalizer creates an I2C connection
+bool fBme;		// true if BME280 is in use
+
+//   The LUX sensor
+BH1750 gBH1750;
+bool fLux;		// true if BH1750 is in use
+
+//   The contact sensors
+bool fHasPower1;
+uint8_t kPinPower1P1;
+uint8_t kPinPower1P2;
+
+cTotalizer gPower1P1;
+cTotalizer gPower1P2;
+
+modbus_t telegram;
+bool fModBusSlave;
+unsigned long u32wait;
+
+//   The flash
+cFlash_AT25SF081 gFlash;
+bool fFlash;
+
+// SCK is D12, which is SERCOM1.3
+// MOSI is D11, which is SERCOM1.0
+// MISO is D10, which is SERCOM1.2
+// Knowing that, you might be able to make sens of the following:
+SPIClass gSPI2(&sercom1, 10, 12, 11, SPI_PAD_0_SCK_3, SERCOM_RX_PAD_2);
+
+//  the LMIC job that's used to synchronize us with the LMIC code
+static osjob_t sensorJob;
+
+// debug flag for throttling sleep prints
+bool g_fPrintedSleeping = false;
+
+// the cycle time to use
+unsigned gTxCycle;
+// remaining before we reset to default
+unsigned gTxCycleCount;
+
+static Arduino_LoRaWAN::ReceivePortBufferCbFn receiveMessage;
+
+static inline void powerOn(void)
+	{
+	pinMode(kPowerOn, OUTPUT);
+	digitalWrite(kPowerOn, HIGH);
+	}
+
+static inline void powerOff(void)
+	{
+	pinMode(kPowerOn, INPUT);
+	digitalWrite(kPowerOn, LOW);
+	}
+
+/****************************************************************************\
+|
+|	Code.
+|
+\****************************************************************************/
+
+/*
+
+Name:	setup()
+
+Function:
+	Arduino setup function.
+
+Definition:
+	void setup(
+		void
+		);
+
+Description:
+	This function is called by the Arduino framework after
+	basic framework has been initialized. We initialize the sensors
+	that are present on the platform, set up the LoRaWAN connection,
+	and (ultimately) return to the framework, which then calls loop()
+	forever.
+
+Returns:
+	No explicit result.
+
+*/
+
+void setup(void)
+	{
+	gCatena.begin();
+	powerOn();
+	setup_platform();	// setup platform
+	setup_flash();		// set up flash
+	setup_sensors();	// set up in-built sensors
+	setup_modbus();		// set up Modbus Master
+
+	/* for STM32 devices, we need wider tolerances, it seems */
+#if defined(ARDUINO_ARCH_STM32)
+	LMIC_setClockError(10 * 65536 / 100);
+#endif
```

Comment:
You can delete this code because it is not STM32 sketch. Latest C-A-P support STM32 clock calibration.